microprofile/connectors/kafka/src/main/java/io/helidon/microprofile/connectors/kafka/BasicKafkaConsumer.java

```diff
@@ -0,0 +1,206 @@
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+/**
+ * Basic Kafka consumer covering basic use-cases.
+ * Configurable by Helidon {@link io.helidon.config.Config Config},
+ * For more info about configuration see {@link KafkaConfigProperties}
+ *
+ * @param <K> Key type
+ * @param <V> Value type
+ * @see KafkaConfigProperties
+ * @see io.helidon.config.Config
+ */
+class BasicKafkaConsumer<K, V> implements Closeable {
+
+    private static final Logger LOGGER = Logger.getLogger(BasicKafkaConsumer.class.getName());
+    private static final String POOL_TIMEOUT = "pool.timeout";
+    private static final String PERIOD_EXECUTIONS = "period.executions";
+    private final KafkaConfigProperties properties;
+    private final Config config;
+    private final Lock taskLock = new ReentrantLock();
+    private final PartitionsAssignedLatch partitionsAssignedLatch = new PartitionsAssignedLatch();
+    private final List<String> topicNameList;
+    // It is not thread safe. It needs to be closed in the same thread it reads events.
+    // We need to keep the reference here to be able to wake up from pooling when shuting down
+    private final KafkaConsumer<K, V> consumer;
+    private final ScheduledExecutorService scheduler;
+
+    /**
+     * Kafka consumer created from {@link io.helidon.config.Config config}
+     * see configuration {@link KafkaConfigProperties example}.
+     *
+     * @param config Helidon {@link io.helidon.config.Config config}
+     * @param scheduler Helidon {@link java.util.concurrent.ScheduledExecutorService scheduler}
+     */
+    BasicKafkaConsumer(Config config, ScheduledExecutorService scheduler) {
+        this.config = config;
+        this.properties = new KafkaConfigProperties(config);
+        this.topicNameList = properties.getTopicNameList();
+        this.consumer = new KafkaConsumer<>(properties);
+        this.scheduler = scheduler;
+    }
+
+    /**
+     * Create publisher builder.
+     *
+     * @return {@link org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder}
+     */
+    public PublisherBuilder<? extends Message<?>> createPushPublisherBuilder() {
+        return ReactiveStreams.fromPublisher(new BasicPublisher<K, V>(subscriber -> {
+            subscriber.onSubscribe(new Subscription() {
+                @Override
+                public void request(long n) {
+                    // Pushing Kafka consumer doesn't support requests.
+                }
+
+                @Override
+                public void cancel() {
+                    BasicKafkaConsumer.this.close();
+                    LOGGER.log(Level.FINE, "Subscription cancelled.");
+                }
+            });
+            consumer.subscribe(topicNameList, partitionsAssignedLatch);
+            scheduler.scheduleAtFixedRate(new BackPressureLayer(subscriber,
+                    config.get(POOL_TIMEOUT).asLong().asOptional().orElseGet(() -> 50L)), 0,
```

Comment:
If you return a constant from `Optional.orElseGet`, then use `Optional.orElse`

Reply:
Done